Source/JavaScriptCore/dfg/DFGLoopUnrollingPhase.cpp

```diff
@@ -510,7 +510,10 @@ class LoopUnrollingPhase : public Phase {
         }
 
         // Done clone.
-        m_blockInsertionSet.execute();
+        if (!m_blockInsertionSet.execute()) {
+            m_graph.invalidateCFG();
+            m_graph.dethread();
```

Comment:
If you do the `tail->appendNode(m_graph, SpecNone, Jump, tailTerminalOrigin, OpInfo(next));` via the insertion set it should do this for you, I think.